test/framework/resource/k8s/jobs/manager.go

```diff
@@ -39,6 +43,44 @@ type defaultManager struct {
 	k8sClient client.Client
 }
 
+func (m *defaultManager) CreateJobAndWaitForJobToRun(ctx context.Context,
+	jobs *batchV1.Job) (*batchV1.Job, error) {
+	err := m.k8sClient.Create(ctx, jobs)
+	if err != nil {
+		return nil, err
+	}
+
+	// Wait till the cache is refreshed
+	time.Sleep(utils.PollIntervalShort)
```

Comment:
we can use PollUntil instead of PollImmediateUntil, to avoid the sleep here :D

Reply:
Nice! That's much cleaner. Replaced to PollUntil everywhere.